data_steward/analytics/table_metrics/weekly_scripts/erroneous_dates.py

```diff
@@ -0,0 +1,583 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.3.0
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# ### This notebook is intended to show the percentage of rows where there are 'erroneous dates' in the 6 canonical tables. The 6 canonical tables are as follows:
+# - Condition Occurrence
+# - Procedure Occurrence
+# - Visit Occurrence
+# - Drug Exposure
+# - Measurement
+# - Observation
+#
+# ### Erroneous dates are those that precede 1900 for the observation table or precede 1980 for all other tables
+
+# +
+from google.cloud import bigquery
+
+# %reload_ext google.cloud.bigquery
+
+client = bigquery.Client()
+
+# %load_ext google.cloud.bigquery
+
+# +
+from notebooks import parameters
+DATASET = parameters.LATEST_DATASET
+
+print("Dataset to use: {DATASET}".format(DATASET = DATASET))
+
+# +
+#######################################
+print('Setting everything up...')
+#######################################
+
+import warnings
+
+warnings.filterwarnings('ignore')
+import pandas as pd
+import matplotlib.pyplot as plt
+# %matplotlib inline
+import os
+
+
+plt.style.use('ggplot')
+pd.options.display.max_rows = 999
+pd.options.display.max_columns = 999
+pd.options.display.max_colwidth = 999
+
+
+def cstr(s, color='black'):
+    return "<text style=color:{}>{}</text>".format(color, s)
+
+
+print('done.')
+# -
+
+cwd = os.getcwd()
+cwd = str(cwd)
+print("Current working directory is: {cwd}".format(cwd=cwd))
+
+# +
+dic = {
+    'src_hpo_id': [
+        "saou_uab_selma", "saou_uab_hunt", "saou_tul", "pitt_temple",
+        "saou_lsu", "trans_am_meyers", "trans_am_essentia", "saou_ummc",
+        "seec_miami", "seec_morehouse", "seec_emory", "uamc_banner", "pitt",
+        "nyc_cu", "ipmc_uic", "trans_am_spectrum", "tach_hfhs", "nec_bmc",
+        "cpmc_uci", "nec_phs", "nyc_cornell", "ipmc_nu", "nyc_hh",
+        "ipmc_uchicago", "aouw_mcri", "syhc", "cpmc_ceders", "seec_ufl",
+        "saou_uab", "trans_am_baylor", "cpmc_ucsd", "ecchc", "chci", "aouw_uwh",
+        "cpmc_usc", "hrhc", "ipmc_northshore", "chs", "cpmc_ucsf", "jhchc",
+        "aouw_mcw", "cpmc_ucd", "ipmc_rush", "va", "saou_umc"
```

Comment:
For these, try to create your own query to fetch data from the table `lookup_tables.hpo_site_id_mappings`

Reply:
This script does not leverage Google bigquery in any sense (I don't even run it in Jupyter noetbooks). I could hypothetically refactor it accordingly but that would fall in line with EDQ-438, which is 'low priority'.